Sample Application/SampleApplication/Screens/Authentication/AuthenticationView.swift

```diff
@@ -0,0 +1,116 @@
+                .font(Font(DesignSystem.AuthenticationScreenDesign.titleFont))
+                .multilineTextAlignment(.center)
+            Spacer(minLength: Constants.Size.spacer)
+
+            switch viewModel.viewState {
+            case .initial:
+                viewGroup(shouldCircleSpin: false,
+                          shouldUnlock: false,
+                          buttonText: Constants.Strings.authenticate,
+                          buttonAction: viewModel.authenticateButtonTapped)
+
+            case .loading:
+                viewGroup(shouldCircleSpin: true,
+                          shouldUnlock: false,
+                          buttonText: Constants.Strings.authenticating)
+
+            case .loaded:
+                viewGroup(shouldCircleSpin: false,
+                          shouldUnlock: true,
+                          buttonText: Constants.Strings.authenticated,
+                          animationsCompleted: {
+                    dismiss()
+                })
+            }
+        }
+        .onAppear {
+            updateViewModel()
+        }
+    }
+
+    @ViewBuilder
+    private func viewGroup(shouldCircleSpin: Bool,
+                           shouldUnlock: Bool,
+                           buttonText: String,
+                           buttonAction: (() -> Void)? = nil,
+                           animationsCompleted: (() -> Void)? = nil) -> some View {
+
+        ZStack {
+            SpinningCircleView(isAnimating: $isCircleSpinning)
+                .onAppear {
+                    isCircleSpinning = shouldCircleSpin
+                }
+
+            LockView(isUnlocked: $isUnlocked,
+                     animationsCompleted: animationsCompleted)
+            .onAppear {
+                isUnlocked = shouldUnlock
+            }
+        }
+
+        Button(action: buttonAction ?? {}) {
+            Text(buttonText)
+                .frame(maxWidth: .infinity, maxHeight: .infinity)
+        }.disabled(buttonAction == nil)
+            .frame(height: 50, alignment: .center)
+            .frame(maxWidth: .infinity)
+            .background(Color(DesignSystem.AuthenticationScreenDesign.buttonBackgroundColor))
+            .foregroundColor(Color(DesignSystem.AuthenticationScreenDesign.buttonTextColor))
+            .font(Font(DesignSystem.AuthenticationScreenDesign.buttonFont))
+            .cornerRadius(25)
+            .padding(32)
+    }
+
+    private func updateViewModel() {
+        viewModel.authenticationRecord = authenticationRecord
+    }
+}
+
+extension AuthenticationView {
+    enum Constants {
+        enum Strings {
+            static let authenticationRequired = "Authentication \n Required"
+            static let authenticate = "Authenticate"
+            static let authenticating = "Authenticating..."
+            static let authenticated = "Authenticated"
+        }
+
+        enum Size {
+            static let spacer: CGFloat = 35
+        }
```

Comment:
These could probably go into the `DesignConstants`, but not necessarily now. Can just add to a generic task of reviewing, part of the next stage of work.